internal/controller/resourceset_controller.go

```diff
@@ -550,6 +550,12 @@ func (r *ResourceSetReconciler) copyResources(ctx context.Context,
 				if err := kubeClient.Get(ctx, sourceName, secret); err != nil {
 					return fmt.Errorf("failed to copy data from Secret/%s: %w", source, err)
 				}
+				if secret.Type == "" {
+					secret.Type = corev1.SecretTypeOpaque
+				}
+				if err := unstructured.SetNestedField(objects[i].Object, string(secret.Type), "type"); err != nil {
+					return fmt.Errorf("failed to copy type from Secret/%s: %w", source, err)
+				}
```

Comment:
We need to set the type only if it's not present in the ResourceSet https://github.com/controlplaneio-fluxcd/d2-fleet/blob/main/tenants/apps.yaml#L41-L48

So instead of checking the original secret, check the unstructured object first.

Reply:
Fixed, thanks!